src/cleanfigure.m

```diff
@@ -449,6 +451,47 @@ function simplifyLine(meta, handle, targetResolution)
     removeData(meta, handle, id_remove);
 end
 % =========================================================================
+function simplifyStairs(meta, handle)
+   % Some data might not lead to a new step in the stair, e.g.
+   % [(x_1, y_1), (x_2, y_1), ... (x_k, y_1), (x_{k+1} y_2)] 
+   % equals [(x_1, y_1), (x_{k+1} y_2)]
+
+    % Extract the data
+    xData = get(handle, 'XData');
+    yData = get(handle, 'YData');
+
+    % Dont do anything if the data is empty
+    if isempty(xData) || isempty(yData)
+        return;
+    end
+
+    % Check for identic data points
+    xNoDiff      = [false, (diff(xData) == 0)];
+    yNoDiff      = [false, (diff(yData) == 0)];
+
+    % Do not remove the last data point
+    xNoDiff(end) = false;
+    yNoDiff(end) = false;
+
+    % Identic data might lead to a visible result if the sign in the other
+    % dimension changes , e.g. the sequence  [0, 1], [0, -1], [0, 2]
+    % Therefore check for a sign change in the difference between data
+    % points (The sign changes if diff(sign)~=0)
+    xIsMonotone  = [true, diff(sign(diff(xData)))==0, true];
```

Comment:
I think we better extract this to a function.
Moreover, I have my doubts about the correctness (either of the code or my understanding of this)

As I remember, we need to check that the whole sequence is monotonous: `x(i) <= x(j) if i < j for each i != j` (or `x(i) >= x(j) ... `)
And this function doesn't seem to do that, it rather checks for locally strict monotonicity, i.e. `x(i) < x(i+1) for each i` (or `x(i) > x(i) ...`).

Perhaps we should add some documentation in the code, e.g. some ASCII plots or data sequences which highlights what kind of points can be removed (somewhere before we actually compute anything).
```matlab
% E.g. in the plot below, the data points are indicated by `o` and `x`. We can remove the `x` points without altering the lines. So this means, either:
%  - one of the coordinates is not changing
%  - ...
%
%       x--o--+
%       |     |
% o--x--o     o--o
%
```